daemon/inertia/auth/util.go

```diff
@@ -0,0 +1,15 @@
+package auth
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"io"
+)
+
+func generateSessionID() string {
+	b := make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, b); err != nil {
+		return ""
```

Comment:
Seems like the fallback in the (extremely rare) case that we fail to read from `rand.Reader` should be to throw some sort of error, rather than defaulting to setting the session ID to the empty string.

Reply:
I was a bit hesitant to add a `error` return on such a trivial function that should practically never fail, since `rand.Reader` should be safe to rely on and `b` is allocated within the same function and not passed in from elsewhere, but I suppose it's just a few extra lines of `err` checking

![](https://pbs.twimg.com/media/DCIF7-2W0AEAv9c.jpg)